### PR TITLE
ceph: allow user to add labels to pods

### PR DESCRIPTION
**Pull request by galexrt**
**Description of your changes:**
This allows users to specify additional labels to be added to the components
created by the Rook Ceph operator.

Signed-off-by: Alexander Trost <galexrt@googlemail.com>

**This PR will be a draft till #6082 is merged.**

**Which issue is resolved by this Pull Request:**
Relates to #6050

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [x] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.

// known issue
[skip ci]

### CODE REVIEW

File: Documentation/ceph-cluster-crd.md [hunk 1]
```diff
@@ -377,16 +378,16 @@ A Drive Group is defined by a name, a Ceph Drive Group spec, and a Rook placemen
   (Optional) Default is no placement criteria, which matches all untainted nodes.
   The syntax is the same as for [other placement configuration](#placement-configuration-settings).
 
-### Annotations Configuration Settings
+### Annotations / Labels Configuration Settings
```
Comment (travisn) [inline Documentation/ceph-cluster-crd.md:381]: It's nice to keep titles short IMO.

```suggestion
### Annotations and Labels
```

File: pkg/operator/ceph/cluster/osd/provision_spec.go [hunk 2]
```diff
@@ -155,6 +155,7 @@ func (c *Cluster) provisionPodTemplateSpec(osdProps osdProperties, restart v1.Re
 	}
 
 	cephv1.GetOSDAnnotations(c.spec.Annotations).ApplyToObjectMeta(&podMeta)
+	cephv1.GetOSDLabels(c.spec.Labels).ApplyToObjectMeta(&podMeta)
```
Comment (travisn) [inline pkg/operator/ceph/cluster/osd/provision_spec.go:158]: Instead of OSD labels, we should have a separate category for `prepareosd` labels

Comment (galexrt) [reply pkg/operator/ceph/cluster/osd/provision_spec.go:158]: Sure, sounds good.

File: pkg/apis/ceph.rook.io/v1/keys.go [hunk 3]
```diff
@@ -21,8 +21,9 @@ import (
 )
 
 const (
-	KeyMon     rook.KeyType = "mon"
-	KeyMgr     rook.KeyType = "mgr"
-	KeyOSD     rook.KeyType = "osd"
-	KeyCleanup rook.KeyType = "cleanup"
+	KeyMon        rook.KeyType = "mon"
+	KeyMgr        rook.KeyType = "mgr"
+	KeyOSDPrepare rook.KeyType = "osdprepare"
```
Comment (travisn) [inline pkg/apis/ceph.rook.io/v1/keys.go:26]: Let's be consistent with the resources that use: `ResourcesKeyPrepareOSD = "prepareosd"`. Separately, we should really merge the resources and key types so we don't have these differences.
```suggestion
	KeyOSDPrepare rook.KeyType = "prepareosd"
```

File: cluster/examples/kubernetes/ceph/cluster.yaml [hunk 4]
```diff
@@ -139,7 +139,13 @@ spec:
...
 #    cleanup:
 # If no mgr annotations are set, prometheus scrape annotations will be set by default.
-#   mgr:
+#    mgr:
+  labels:
+#    all:
+#    mon:
+#    osd:
+#    cleanup:
+#    mgr:
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/cluster.yaml:148]: need another line for `# prepareosd`

**Pull request merged**